packages/api/amplify_api/lib/src/api_plugin_impl.dart

```diff
@@ -60,9 +62,17 @@ class AmplifyAPIDart extends AmplifyAPI {
   /// The registered [APIAuthProvider] instances.
   final Map<APIAuthorizationType, APIAuthProvider> _authProviders = {};
 
+  final StreamController<ApiHubEvent> _hubEventController =
+      StreamController<ApiHubEvent>.broadcast();
+
   /// Subscription options
   final GraphQLSubscriptionOptions? subscriptionOptions;
 
+  @override
+  Future<void> reset() async {
+    await _hubEventController.close();
```

Comment:
I would do something like this to ensure this works correctly given what we saw in the tests

```suggestion
    for (final bloc in _webSocketBlocPool.values) {
       bloc.add(ShutdownEvent());
    }
    await _Future.wait(_webSocketBlocPool.values.map((bloc) => bloc.done));
    await _hubEventController.close();
```